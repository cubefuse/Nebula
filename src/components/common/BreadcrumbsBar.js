import React from "react";
import { Link, withRouter } from "react-router-dom";
import { Breadcrumb } from "antd";

/**
 * App Breadcrumbs bar.
 */
class BreadcrumbsBar extends React.Component {
  constructor(props) {
    super(props);

    // A map of URLs with custom breadcrumb name mappings.
    this.breadcrumbNameMap = {
      "/get": "Download",
      "/404": "404 Not Found"
    };
  }

  render() {
    const pathSnippets = this.props.location.pathname.split("/").filter(i => i);

    const breadcrumbSubItems = pathSnippets.map((_, index) => {
      const url = `/${pathSnippets.slice(0, index + 1).join("/")}`;
      return (
        <Breadcrumb.Item key={url}>
          <Link to={url}>{this.breadcrumbNameMap[url]}</Link>
        </Breadcrumb.Item>
      );
    });

    const breadcrumbItems = [
      <Breadcrumb.Item key="home">
        <Link to="/">Home</Link>
      </Breadcrumb.Item>
    ].concat(breadcrumbSubItems);

    return (
      <Breadcrumb style={{ margin: "16px 0" }}>{breadcrumbItems}</Breadcrumb>
    );
  }
}

export default withRouter(BreadcrumbsBar);
