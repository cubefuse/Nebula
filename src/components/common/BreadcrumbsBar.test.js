import React from "react";
import "jest-dom/extend-expect";
import BreadcrumbsBar from "./BreadcrumbsBar";
import TestHelpers from "../../util/test-helpers";

test("can render with Router", () => {
  const node = TestHelpers.renderWithRouter(<BreadcrumbsBar />);
  expect(node).toMatchSnapshot();
});
