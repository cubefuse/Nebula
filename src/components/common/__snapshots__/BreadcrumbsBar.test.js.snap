// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can render with Router 1`] = `
Object {
  "container": <div>
    <div
      class="ant-breadcrumb"
      style="margin: 16px 0px;"
    >
      <span>
        <span
          class="ant-breadcrumb-link"
        >
          <a
            href="#/"
          >
            Home
          </a>
        </span>
        <span
          class="ant-breadcrumb-separator"
        >
          /
        </span>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "getAllByAltText": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
